modules/mauth-authenticator-apachehttp/src/main/java/com/mdsol/mauth/apache/HttpClientPublicKeyProvider.java

```diff
@@ -39,25 +37,20 @@
   private final PublicKeyResponseHandler publicKeyResponseHandler;
 
   private LoadingCache<UUID, PublicKey> publicKeyCache;
+  private Long ttl;
 
   public HttpClientPublicKeyProvider(AuthenticatorConfiguration configuration, Signer signer) {
     this.configuration = configuration;
     this.signer = signer;
     this.httpclient = HttpClients.createDefault();
     this.publicKeyResponseHandler = new PublicKeyResponseHandler();
-    setupCache(configuration.getTimeToLive());
   }
 
-  private void setupCache(long timeToLiveInSeconds) {
+  private void setupCache() {
     publicKeyCache =
-        CacheBuilder.newBuilder()
-            .expireAfterAccess(timeToLiveInSeconds, TimeUnit.SECONDS)
-            .build(new CacheLoader<UUID, PublicKey>() {
-              @Override
-              public PublicKey load(UUID appUUID) throws Exception {
-                return getPublicKeyFromEureka(appUUID);
-              }
-            });
+      Caffeine.newBuilder()
+        .expireAfterAccess(ttl, TimeUnit.SECONDS)
+        .build(this::getPublicKeyFromEureka);
```

Comment:
This should probably be renamed `getPublicKeyFromMauth`, mauth clients shouldn't depend on eureka at all.

Reply:
This was on the old code but yes I do agree to change it. Already changed in my new commit.